gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/NewDagManager.java

```diff
@@ -0,0 +1,575 @@
+    return this.resumingDags;
+  }
+
+  public Map<Dag.DagNode<JobExecutionPlan>, Dag<JobExecutionPlan>> getJobToDag() {
+    return this.jobToDag;
+  }
+
+  public Map<String, Dag.DagNode<JobExecutionPlan>> getDagNodes() {
+    return this.dagNodes;
+  }
+
+  public Map<String, LinkedList<Dag.DagNode<JobExecutionPlan>>> getDagToJobs() {
+    return this.dagToJobs;
+  }
+
+  public Map<String, Long> getDagToSLA() {
+    return this.dagToSLA;
+  }
+
+  public Set<String> getFailedDagIds() {
+    return this.failedDagIds;
+  }
+
+  public DagStateStore getFailedDagStateStore() {
+    return this.failedDagStateStore;
+  }
+
+  public DagStateStore getDagStateStore() {
+    return this.dagStateStore;
+  }
+
+  public Integer getNumThreads() {
+    return this.numThreads;
+  }
+
+  public JobStatusRetriever getJobStatusRetriever() {
+    return this.jobStatusRetriever;
+  }
+
+  public UserQuotaManager getQuotaManager() {
+    return this.quotaManager;
+  }
+
+  public Optional<Timer> getJobStatusPolledTimer() {
+    return this.jobStatusPolledTimer;
+  }
+
+  public Optional<EventSubmitter> getEventSubmitter() {
+    return this.eventSubmitter;
+  }
+
+  public DagManagerMetrics getDagManagerMetrics() {
+    return this.dagManagerMetrics;
+  }
+
+  public AtomicLong getOrchestrationDelay() {
+    return this.orchestrationDelay;
+  }
+
+  public DagProcessingEngine getDagProcessingEngine() {
+    return this.dagProcessingEngine;
+  }
+
+  public Optional<DagActionStore> getDagActionStore() {
+    return this.dagActionStore;
+  }
+
+  /**
+   * Action to be performed on a {@link Dag}, in case of a job failure. Currently, we allow 2 modes:
+   * <ul>
+   *   <li> FINISH_RUNNING, which allows currently running jobs to finish.</li>
+   *   <li> FINISH_ALL_POSSIBLE, which allows every possible job in the Dag to finish, as long as all the dependencies
+   *   of the job are successful.</li>
+   * </ul>
+   */
+  public enum FailureOption {
+    FINISH_RUNNING("FINISH_RUNNING"),
+    CANCEL("CANCEL"),
+    FINISH_ALL_POSSIBLE("FINISH_ALL_POSSIBLE");
+
+    private final String failureOption;
+
+    FailureOption(final String failureOption) {
+      this.failureOption = failureOption;
+    }
+
+    @Override
+    public String toString() {
+      return this.failureOption;
+    }
+  }
+
+  private final Map<String, Dag<JobExecutionPlan>> dags = new HashMap<>();
+  private final Map<String, Dag<JobExecutionPlan>> resumingDags = new HashMap<>();
+  private static final long DAG_FLOW_STATUS_TOLERANCE_TIME_MILLIS = TimeUnit.MINUTES.toMillis(5);
+
+
+  private final Map<Dag.DagNode<JobExecutionPlan>, Dag<JobExecutionPlan>> jobToDag = new HashMap<>();
+  private final Map<String, Dag.DagNode<JobExecutionPlan>> dagNodes = new HashMap<>();
+  private final Map<String, LinkedList<Dag.DagNode<JobExecutionPlan>>> dagToJobs = new HashMap<>();
+  final Map<String, Long> dagToSLA = new HashMap<>();
+  DagManager.DagManagerThread[] dagManagerThreads;
```

Comment:
`DagManagement` has api to use `DagManagementStateStore` and we should remove these in-memory references